core/io/file_access_compressed.cpp

```diff
@@ -273,7 +275,8 @@ uint8_t FileAccessCompressed::get_8() const {
 		if (read_block < read_block_count) {
 			//read another block of compressed data
 			f->get_buffer(comp_buffer.ptrw(), read_blocks[read_block].csize);
-			Compression::decompress(buffer.ptrw(), read_blocks.size() == 1 ? read_total : block_size, comp_buffer.ptr(), read_blocks[read_block].csize, cmode);
+			int ret = Compression::decompress(buffer.ptrw(), read_blocks.size() == 1 ? read_total : block_size, comp_buffer.ptr(), read_blocks[read_block].csize, cmode);
+			ERR_FAIL_COND_V_MSG(ret == -1, 0, "Compressed file is corrupt.");
```

Comment:
This shadows `ret` which is actually the decompression result.

Reply:
Fixed, I picked `total` as a more sensible name than `err` for this one.